src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -50,6 +52,31 @@ internal static class TypeHelpers
                 // Get the state from DebuggerBrowsableAttributes for the members of the current type.
                 var browsableState = DkmClrType.Create(appDomain, type).GetDebuggerBrowsableAttributeState();
 
+                // Disable favorites if any of the members have a browsable state of RootHidden
+                if (supportsFavorites && browsableState != null)
+                {
+                    foreach (var browsableStateValue in browsableState.Values)
+                    {
+                        if (browsableStateValue == DkmClrDebuggerBrowsableAttributeState.RootHidden)
+                        {
+                            supportsFavorites = false;
+                            break;
+                        }
+                    }
+                }
+
+                // Get the favorites information if it is supported.
+                Dictionary<string, object> favoritesMemberNames = null;
```

Comment:
>Dictionary<string, object> [](start = 16, length = 26)

it seems that a hashset is enough.

Reply:
It is, but unfortunately this still has to build for .net 2.0 which does not have it.

Reply:
Wow.. interesting! 
And should this binary work under .net 2.0 as well?

If so, could you please then add a comment about this and the usage of Dictionary for the purpose?

---
In reply to: [326297622](https://github.com/dotnet/roslyn/pull/38669#discussion_r326297622) [](ancestors = 326297622)